app/model/lib/submission_process.py

```diff
@@ -286,23 +290,30 @@ def _save_experiments(db_session, submission_form, study):
     compartments_by_name = group_by_unique_name(study.compartments)
 
     for experiment_data in submission.studyDesign['experiments']:
-        experiment_data = copy.deepcopy(experiment_data)
+        experiment_params = copy.deepcopy(experiment_data)
 
-        community_name    = experiment_data.pop('communityName')
-        compartment_names = experiment_data.pop('compartmentNames')
-        bioreplicates     = experiment_data.pop('bioreplicates')
-        perturbations     = experiment_data.pop('perturbations')
+        community_name    = experiment_params.pop('communityName')
+        compartment_names = experiment_params.pop('compartmentNames')
+        bioreplicates     = experiment_params.pop('bioreplicates')
+        perturbations     = experiment_params.pop('perturbations')
+
+        if publicId := experiment_params.pop('publicId', None):
+            experiment = db_session.get(Experiment, publicId)
```

Comment:
An experiment used to have an autoincrementing `id` as a primary key. Migrations in this PR remove that primary key and make the `publicId` a primary key, same as studies and projects.